### ISSUE TITLE
Dose CTC need SOFTMAX?

### ISSUE THREAD

**Issue by nightinwhite** [tensorflow/tensorflow#4248]
It seems CTC dose not need SOFTMAX in many example,so how it works?,and if ctc_loss itself has a SOFTMAX,dose it share SOFTMAX with ctc_greedy_decoder,I mean if there is no SOFTMAX how can I use it to predict ?
Thanks for any help

**Comment by tatatodd** [comment]
We primarily use github issues to track bugs and installation issues.  This is a question better suited for StackOverflow.  Please ask it there and tag it with the `tensorflow` tag.  Thanks!

That said, a cursory search on stackoverflow shows a link that might help:
http://stackoverflow.com/questions/38059247/using-tensorflows-connectionist-temporal-classification-ctc-implementation/38069255#38069255

**Comment by selcouthlyBlue** [comment]
It's in the [documentation](https://www.tensorflow.org/versions/r0.12/api_docs/python/nn/connectionist_temporal_classification__ctc_) of `tf.nn.ctc_loss`.

> This class performs the softmax operation for you, so inputs should be e.g. linear projections of outputs by an LSTM.
